tests/console/validate_lvm_with_raid1.pm

```diff
@@ -0,0 +1,88 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple LVM partition validation
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "y2_module_consoletest";
+use testapi;
+use utils;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
+
+my $config;
+my @expectedDevs;
+my $expected_num_devs;
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+
+    $config            = get_test_data();
+    @expectedDevs      = (keys %{$config->{disks}});
+    $expected_num_devs = @expectedDevs;
+
+    _lvm_partitioning();
+    _raid1_partitioning();
+    $self->_raid_can_boot_with_deleted_hdd();
+}
+
+sub _lvm_partitioning {
+    my $activelvm = script_output q[lvscan | awk '{print $2}' | sed s/\'//g];
+    assert_equals $config->{lvm}->{lvpath}, $activelvm, "lv name is not the same. Device might not be Active";
+    my $pvname = script_output q[pvscan -s | grep '/dev'];
+    assert_equals $config->{lvm}->{pvname}, $pvname, "pv name is not the same.";
+}
+
+sub _raid1_partitioning {
+    record_info("print info", $config->{raid1}->{name});
+    assert_script_run 'mdadm --detail /dev/' . $config->{raid1}->{name};
+
+    assert_script_run 'grep \'md0 : active ' . $config->{raid1}->{level} . '\' /proc/mdstat';
+
+    my $active_devs = script_output("mdadm --detail /dev/md0 |grep \"Active Devices\" |awk '{ print \$4 }'");
+    assert_equals($active_devs, $expected_num_devs, "Active devices are differ");
+    for (@expectedDevs) {
+        my $line = "mdadm --detail /dev/md0 | awk '{ if((/$_/) && (\$5 ~ /active/) && (\$6 ~ /sync/)) {print}}'";
+        assert_not_null($line, "$_ not active");
```

Comment:
Looks like you wanted to call `script_output` here. If not it's non-sense to define string and then check that it's not undef =)

Reply:
my bad. fixed

Reply:
you could add `or not sync` in the message as there are possible status like for example `spare rebuilding`. Here it is similar to other cases, that you could set expectation of sync and active outside the code in the `test_data` to avoid `assert_not_null` that is quite peculiar, but feel free to leave it as it is as the awk is quite readable and the optional message is also enough descriptive.